typescript/sdk/src/warp/WarpCore.ts

```diff
@@ -197,13 +304,109 @@ export class WarpCore {
     return transactions;
   }
 
+  /**
+   * Fetch local and interchain fee estimates for a remote transfer
+   */
+  async estimateTransferRemoteFees({
+    originToken,
+    destination,
+    sender,
+    senderPubKey,
+  }: {
+    originToken: IToken;
+    destination: ChainNameOrId;
+    sender: Address;
+    senderPubKey?: HexString;
+  }): Promise<WarpCoreFeeEstimate> {
+    this.logger('Fetching remote transfer fee estimates');
+
+    // First get interchain gas quote (aka IGP quote)
+    // Start with this because it's used in the local fee estimation
+    const interchainQuote = await this.getInterchainTransferFee({
+      originToken,
+      destination,
+    });
+
+    const originMetadata = this.multiProvider.getChainMetadata(
+      originToken.chainName,
+    );
+    // If there's no native token, we can't represent local gas
+    if (!originMetadata.nativeToken)
+      throw new Error(`No native token found for ${originMetadata.name}`);
+
+    // Next, get the local gas quote
+    const localFee = await this.getLocalTransferFee({
+      originToken,
+      destination,
+      sender,
+      senderPubKey,
+      interchainFee: interchainQuote,
+    });
+
+    // Get the local gas token. This assumes the chain's native token will pay for local gas
+    // This will need to be smarter if more complex scenarios on Cosmos are supported
+    const localGasToken = Token.FromChainMetadataNativeToken(originMetadata);
+    const localQuote = localGasToken.amount(localFee.fee);
+
+    return {
+      interchainQuote,
+      localQuote,
+      localDetails: localFee,
+    };
+  }
+
+  /**
+   * Computes the max transferrable amount of the from the given
+   * token balance, accounting for local and interchain gas fees
+   */
+  async getMaxTransferAmount({
+    balance,
+    destination,
+    sender,
+    senderPubKey,
+    feeEstimate,
+  }: {
+    balance: TokenAmount;
+    destination: ChainNameOrId;
+    sender: Address;
+    senderPubKey?: HexString;
+    feeEstimate?: WarpCoreFeeEstimate;
+  }): Promise<TokenAmount> {
+    const originToken = balance.token;
+
+    if (!feeEstimate) {
```

Comment:
I would remove this optional param and always do this estimation here

Reply:
I added this specifically to avoid re-querying for the estimate when we just got it. Unless we implement a query caching layer, I'd like to keep this. The Max button feels slow already